grakn-graql/src/main/java/ai/grakn/graql/internal/pattern/property/SubProperty.java

```diff
@@ -120,7 +120,7 @@ public Atomic mapToAtom(VarAdmin var, Set<VarAdmin> vars, ReasonerQuery parent)
         VarName varName = var.getVarName();
         VarAdmin typeVar = this.getSuperType();
         VarName typeVariable = typeVar.isUserDefinedName() ?
-                typeVar.getVarName() : varName.map(name -> name + "-sub-" + UUID.randomUUID().toString());
+                typeVar.getVarName() : varName.map(name -> name + "-" + getName() + "-" + UUID.randomUUID().toString());
```

Comment:
and this are the same. Is this intentional? Since they all property maybe push this line up into a parent and access it from there?